CI should use a locked version of Deno

Note for example that the release PR for 0.95.0 (#869) is tested with 1.9.1 which is one behind its supposed companion version of CLI (listed [here](https://github.com/denoland/deno_website2/blob/923ac3cd7335203afcf5c64cdc3373dfbc0e52da/versions.json#L108)). We need to install a specific version of Deno in CI. This should be bumped on an std release, which should wait for its companion CLI version to release and publish first.

---

On a separate note, I've thought more about what purpose the canary tests serve and I don't think they have any business being here. So far what they've done is allow us to see breaking changes in CLI (via random non-deterministic breakages in PRs here) so we can go back and revert them. It's much more sensible to have https://github.com/denoland/deno/pull/9801 for that instead.

What they don't do, is allow us to make adaptive changes in std for intentional breaking changes so std is always compatible with `--canary`. Because when that std change is released to users, there is a fresh CLI release which outdates that `--canary` version and it entirely suffices to adapt when bumping to that.

To add on canary tests within CI, it seems possible that canary builds are not finished yet when PR is submitted yet resulting in these kinds of errors:

> Looking up latest version
Found latest version 5214acd3d9dec56ee159544f0f6bf9834a62c097
Checking https://dl.deno.land/canary/5214acd3d9dec56ee159544f0f6bf9834a62c097/deno-x86_64-apple-darwin.zip
Download could not be found, aborting
Error: Process completed with exit code 1.

--- a/README.md
+++ b/README.md
@@ -90,7 +90,8 @@ For contributions to the Node compatibility library please check the
 
 _About CI checks_:
 
-We currently have 9 checks on CI. You need to pass the following 6 checks for your PR to be accepted:
+We currently have 9 checks on CI. You need to pass the following 6 checks for
+your PR to be accepted:
 
 - test with Deno 1.x on Windows
 - test with Deno 1.x on Linux
@@ -99,7 +100,9 @@ We currently have 9 checks on CI. You need to pass the following 6 checks for yo
 - wasm crypto check
 - CLA
 
-You don't need to pass the following 3 checks (These are informative checks for maintainers):
+You don't need to pass the following 3 checks (These are informative checks for
+maintainers):
+
 - test with Deno canary on Windows
 - test with Deno canary on Linux
 - test with Deno canary on macOS

--- a/README.md
+++ b/README.md
@@ -88,6 +88,22 @@ Ensure there is a related issue and it is referenced in the PR text.
 For contributions to the Node compatibility library please check the
 [`std/node` contributing guide](./node/README.md)
 
+_About CI checks_:
+
+We currently have 9 checks on CI. You need to pass the following 6 checks for your PR to be accepted:
+
+- test with Deno 1.x on Windows
+- test with Deno 1.x on Linux
+- test with Deno 1.x on macOS
+- lint
+- wasm crypto check
+- CLA
+
+You don't need to pass the following 3 checks (These are informative checks for maintainers):
+- test with Deno canary on Windows
+- test with Deno canary on Linux
+- test with Deno canary on macOS
+
 _For maintainers_:
 
 To release a new version a tag in the form of `x.y.z` should be added.

